pyramid/config/__init__.py

```diff
@@ -903,6 +921,13 @@ def end(self):
         """
         return self.manager.pop()
 
+    def __enter__(self):
+        self.begin()
+        return self
+
+    def __exit__(self, exc_type, exc_value, exc_traceback):
+        self.end()
```

Comment:
I can't think of any sane way to detect that we are inside another context manager.  Maybe we could add an opt-out parameter to the `Configurator`?  E.g.:

```python

if __name__ == '__main__':
    with Configurator() as outer:

        # do stuff with outer

        with Configurator(commit_on_exit=False) as inner:
            inner.add_route('hello', '/hello/{name}')
            inner.add_view(hello_world, route_name='hello')

        # do more stuff with outer

    app = outer.make_wsgi_app()
    server = make_server('0.0.0.0', 8080, app)
    server.serve_forever()